src/core/src/op/paged_attention.cpp

```diff
@@ -8,6 +8,88 @@
 #include "itt.hpp"
 #include "openvino/op/op.hpp"
 
+namespace {
+
+using namespace ov;
+using namespace ov::element;
+
+// Validates input rank and type for a node input.
+// We consider that dynamic rank/type are always valid case.
+// Empty {} means any rank/type
+inline void input_check(ov::Node* node,
+                        size_t idx,
+                        const std::string& input_name,
+                        const std::set<int64_t>& allowed_ranks,
+                        std::set<Type> allowed_types) {
+    using namespace ov;
+    using namespace ov::element;
+
+    const auto& pshape = node->get_input_partial_shape(idx);
+    const auto& etype = node->get_input_element_type(idx);
+
+    auto rank_check = [&](const PartialShape& ps) {
```

Comment:
For rank check from list of ranks there is `ov::util::is_rank_compatible_any_of' which accept initializer list.

Reply:
done